test-data/unit/check-classes.test

```diff
@@ -894,6 +892,281 @@ a.f = a.f # E: Property "f" defined in "A" is read-only
 a.f.x # E: "int" has no attribute "x"
 [builtins fixtures/property.pyi]
 
+-- Descriptors
+-- -----------
+
+
+[case testAccessingNonDataDescriptor]
+class D():
+    def __get__(self, instance: Any, owner: Any) -> str:
```

Comment:
Yeah, I think I need to add some checking to validate that `__get__` and `__set__` definitions are correctly typed in definitions. Unless you have a pointer on where that should be checked, I'll do some digging and see if there are any other checks of that sort.